javaagent-tooling/src/main/java/io/opentelemetry/javaagent/extension/instrumentation/ActualInstrumentationExtensionImplementation.java

```diff
@@ -68,7 +68,7 @@ AgentBuilder extend(
 
     ElementMatcher.Junction<ClassLoader> moduleClassLoaderMatcher =
         instrumentationModule.classLoaderMatcher();
-    MuzzleMatcher muzzleMatcher = new MuzzleMatcher(instrumentationModule);
+    MuzzleMatcher muzzleMatcher = new MuzzleMatcher(instrumentationModule, helperClassNames);
```

Comment:
is this purpose for passing helperClassNames here just an optimization to avoid calling `getMuzzleHelperClassNames()` multiple times? maybe ideal for `getMuzzleHelperClassNames()` (and other generated methods?) to return constant to avoid that concern (not a suggestion for this PR)

Reply:
Yeah, it's just an optimization. I guess that helper class names could use the same pattern as references (lazy-init field).